lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -17,6 +17,14 @@ export type DashboardActivityFiltersProps = {
   selectedStudentIds: string[];
   onStudentsChange: (newStudentIds: string[]) => void;
   onClearSelection?: () => void;
+
+  /**
+   * Consumers can use this to indicate that there are filters selected.
+   * If this is not provided, it will be computed based on the presence of
+   * items in either one of `selectedCourseIds`, `selectedAssignmentIds` or
+   * `selectedStudentIds`.
+   */
+  hasSelection?: boolean;
```

Comment:
An alternative to having this prop would be to conditionally set `onClearSelection` to `undefined` in the parent component.

Reply:
In this case `onClearSelection` is needed, so that we can still clear assignments and students dropdowns, but together with this prop, to prevent the clear button to appear for the only presence of the courseId

Reply:
To clarify, what I'm thinking of is:

```diff
diff --git a/lms/static/scripts/frontend_apps/components/dashboard/CourseActivity.tsx b/lms/static/scripts/frontend_apps/components/dashboard/CourseActivity.tsx
index a0c6c107b..dec28fcb7 100644
--- a/lms/static/scripts/frontend_apps/components/dashboard/CourseActivity.tsx
+++ b/lms/static/scripts/frontend_apps/components/dashboard/CourseActivity.tsx
@@ -69,6 +69,12 @@ export default function CourseActivity() {
   const title = course.data?.title ?? 'Untitled course';
   useDocumentTitle(title);
 
+  const onClearSelection = () => {
+    // Clear every filter but courses
+    updateFilters({ studentIds: [], assignmentIds: [] });
+  };
+  const hasSelection = assignmentIds.length > 0 || studentIds.length > 0;
+
   return (
     <div className="flex flex-col gap-y-5">
       <div>
@@ -103,11 +109,7 @@ export default function CourseActivity() {
         onAssignmentsChange={assignmentIds => updateFilters({ assignmentIds })}
         selectedStudentIds={studentIds}
         onStudentsChange={studentIds => updateFilters({ studentIds })}
-        onClearSelection={() =>
-          // Clear every filter but courses
-          updateFilters({ studentIds: [], assignmentIds: [] })
-        }
-        hasSelection={assignmentIds.length > 0 || studentIds.length > 0}
+        onClearSelection={hasSelection ? onClearSelection : undefined}
       />
       <OrderableActivityTable
         loading={assignments.isLoading}
```

Reply:
Oh! That's very clever, and removes the need for another prop.

I'll do that.

Reply:
Done!